### PR TITLE
chore: migrate the Avatar component to be tailwind-compatible

### PR DESCRIPTION
**Pull request by anucreative**
## DESCRIPTION

Breaking changes:

- No more `color` prop. Override color using Tailwind instead (e.g. `<Avatar className="bg-brand-40" />`)
- No more `w` or `h` prop. Override size using Tailwind `size` (e.g. `<Avatar className="size-[100px]" />`)

## SCREENSHOTS / SCREEN RECORDINGS

image

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [x] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [x] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/Avatar/avatar.module.scss [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    &-lg {
+      --avatar-width: 40px;
+    }
+
+    &-xl {
+      --avatar-width: 50px;
+    }
+
+    &-xxl {
+      --avatar-width: 60px;
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/avatar.module.scss]: Transform to rem

File: lib/src/components/Avatar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export type AvatarProps = CreateWuiProps<'div', AvatarOptions>
+
+export const colors = [
+  'secondary-blue',
+  'secondary-green',
+  'secondary-orange',
+  'secondary-pink',
+  'secondary-teal',
+  'secondary-violet',
+] as const
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/index.tsx:18]: This can be on utils only, do not need to export

File: lib/src/components/Avatar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+export interface AvatarOptions {
+  className?: string
+  getInitials?: (name: string) => string
+  name: string
+  size?: Size
+  src?: string
+}
+
+export type AvatarProps = CreateWuiProps<'div', AvatarOptions>
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/index.tsx]: Do not use old types utilities, check other components

File: lib/src/components/Avatar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      <div
+        aria-label={name}
+        className={cx('root', size && `size-${size}`, color && `background-${color}`, className)}
+        ref={ref}
+        role="img"
+      >
+        {src ? (
+          <img alt={name} src={src} />
+        ) : (
+          <Text className="m-0" fontSize="inherit">
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/index.tsx]:
```suggestion
          <Text as="span" className="font-inherit">
```
I'm not sure about font-inherit :D

**Review by theo-mesnil — COMMENTED** [review]
Tailwind prefere to have not dynamic class, i think is better to directly put a tailwind bg-secondary-* on the return of utils like:

```
export const colors = [
  'bg-secondary-blue',
  'bg-secondary-green',
  'bg-secondary-orange',
  'bg-secondary-pink',
  'bg-secondary-teal',
  'bg-secondary-violet',
] as const
```

File: lib/src/components/Avatar/avatar.module.scss [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    p {
+      font-weight: 500;
+      font-size: inherit;
+      color: var(--color-neutral-90);
+    }
+  }
+
+  .size {
+    &-sm {
+      --avatar-width: 20px;
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/avatar.module.scss]: `--width or --avatarWidth or --avatar-witdth`
We need to found the best way to name this var

File: lib/src/components/Avatar/avatar.module.scss [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    height: var(--avatar-width);
+    font-size: calc(var(--avatar-width) / 2.5);
+
+    img {
+      object-fit: cover;
+      width: 100%;
+      height: 100%;
+    }
+
+    p {
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/avatar.module.scss]:
```suggestion
    span or maybe the text class ? 
```

File: lib/src/components/Avatar/docs/examples/texts.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+const Example = () => {
+  return (
+    <>
+      <Avatar name="W" size="sm" />
+      <Avatar name="We" size="md" />
+      <Avatar name="Wel" size="lg" />
+      <Avatar name="Welc" size="xl" />
+      <Avatar name="Welco" size="xxl" />
+      <Avatar className="bg-brand-40 w-[130px] h-[130px] text-[60px]" name="Custom" size="lg" />
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/docs/examples/texts.tsx]:
```suggestion
      <Avatar className="bg-brand-40 size-[130px] text-[60px]" name="Custom" size="lg" />
```

File: lib/src/components/Avatar/avatar.module.scss (hunk 5 above)
Comment (P1X3L) [reply lib/src/components/Avatar/avatar.module.scss]: My goto on this would be to mimic `styled-components` object style notation for css variables that don't exist in our heme.

styled-components object-styles notation is camelCase.

We would have something like :
`width: --avatarWidth`
`--avatarWidth: var(--according-width-from-theme)`

Does this make any sense to you: `camelCase for css variables not coming from theme` & `kebab-case` for those that come from it?

File: lib/src/components/Avatar/avatar.module.scss (hunk 6 above)
Comment (P1X3L) [reply lib/src/components/Avatar/avatar.module.scss]: the Text component class is `wui-text`. We could definitely rely on it

File: lib/src/components/Avatar/docs/examples/texts.tsx (hunk 7 above)
Comment (P1X3L) [reply lib/src/components/Avatar/docs/examples/texts.tsx]: Don't you think it might be good to use `w-(--width-coming-from-theme-using-rem)` tailwind classes?

File: lib/src/components/Avatar/index.tsx (hunk 2 above)
Comment (anucreative) [reply lib/src/components/Avatar/index.tsx:18]: Used by types.ts as well

File: lib/src/components/Avatar/avatar.module.scss (hunk 6 above)
Comment (anucreative) [reply lib/src/components/Avatar/avatar.module.scss]: Have changed the component to output `p`

File: lib/src/components/Avatar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+const cx = classNames(styles)
+
+export interface AvatarOptions {
+  className?: string
+  getInitials?: (name: string) => string
+  name: string
+  size?: Size
+  src?: string
+}
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/index.tsx]: Put on types.ts

File: lib/src/components/Avatar/index.tsx (hunk 3 above)
Comment (theo-mesnil) [reply lib/src/components/Avatar/index.tsx]: and put on types.ts

File: lib/src/components/Avatar/tests/index.test.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import { screen } from '@testing-library/react'
+
+import { render } from '../../../../tests/index'
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/tests/index.test.tsx]:
```suggestion
import { render } from '@tests'
```

**Comment by anucreative** [comment]
> Tailwind prefere to have not dynamic class, i think is better to directly put a tailwind bg-secondary-* on the return of utils like:
> 
> ```
> export const colors = [
>   'bg-secondary-blue',
>   'bg-secondary-green',
>   'bg-secondary-orange',
>   'bg-secondary-pink',
>   'bg-secondary-teal',
>   'bg-secondary-violet',
> ] as const
> ```

I wanted to keep them the same as the theme keys

File: lib/src/components/Avatar/index.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+export const colors = [
+  'secondary-blue',
+  'secondary-green',
+  'secondary-orange',
+  'secondary-pink',
+  'secondary-teal',
+  'secondary-violet',
+] as const
+
+export const Avatar: React.FC<AvatarProps> = forwardRef<'div', AvatarProps>(
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/index.tsx]:
```suggestion
export const Avatar = forwardRef<'div', AvatarProps>(
```

React FC is useless now

File: lib/src/components/Avatar/index.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+
+import { classNames } from '@/utils'
+import { forwardRef } from '@old/System'
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/index.tsx]:
```suggestion
import { forwardRef } from 'react'
```

File: lib/src/components/Avatar/types.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import type { colors } from './index'
+
+export interface AvatarProps {
+  className?: string
+  getInitials?: (name: string) => string
+  name: string
+  size?: Size
+  src?: string
+}
```
Comment (theo-mesnil) [inline lib/src/components/Avatar/types.ts]:
```suggestion
export interface AvatarOptions {
  getInitials?: (name: string) => string
  name: string
  size?: Size
  src?: string
}
```
